src/test/java/com/wikia/webdriver/testcases/commentstests/BlogCommentsTests.java

```diff
@@ -3,29 +3,54 @@
 import com.wikia.webdriver.common.contentpatterns.PageContent;
 import com.wikia.webdriver.common.core.Assertion;
 import com.wikia.webdriver.common.core.annotations.Execute;
-import com.wikia.webdriver.common.core.configuration.Configuration;
 import com.wikia.webdriver.common.core.helpers.User;
-import com.wikia.webdriver.common.properties.Credentials;
+import com.wikia.webdriver.common.logging.PageObjectLogging;
 import com.wikia.webdriver.common.templates.NewTestTemplate;
 import com.wikia.webdriver.pageobjectsfactory.componentobject.minieditor.MiniEditorComponentObject;
-import com.wikia.webdriver.pageobjectsfactory.pageobject.WikiBasePageObject;
 import com.wikia.webdriver.pageobjectsfactory.pageobject.actions.DeletePageObject;
 import com.wikia.webdriver.pageobjectsfactory.pageobject.signup.UserProfilePageObject;
 import com.wikia.webdriver.pageobjectsfactory.pageobject.wikipage.blog.BlogPageObject;
 
+import org.openqa.selenium.TimeoutException;
+import org.testng.annotations.BeforeMethod;
 import org.testng.annotations.Test;
 
+import java.lang.reflect.Method;
+
 @Test(groups = "comments-blogComments")
 public class BlogCommentsTests extends NewTestTemplate {
 
-  Credentials credentials = Configuration.getCredentials();
+  private BlogPageObject blogPage;
+
+  private String userName;
+
+  /**
+   * Setup test environment - navigate to article page
+   */
+  @BeforeMethod(alwaysRun = true)
+  public void initProfilePageAndCheckLoginStatus(Method method) {
+    User user = method.getAnnotation(Execute.class).asUser();
+    UserProfilePageObject userProfile = new UserProfilePageObject(driver).openProfilePage(User.USER.getUserName(), wikiURL);
+    userProfile.clickOnBlogTab();
+    blogPage = userProfile.openFirstPost();
+
+    if (user != User.ANONYMOUS) {
+      userName = user.getUserName();
+
+      // SUS-1737: In BlogComments tests sometimes user is not authenticated properly
+      // Let's check for this, and if this happens, log the situation and log in again
+      try {
+        blogPage.verifyUserLoggedIn(user);
+      } catch (TimeoutException e) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar-s4/coding_rules#rule_key=squid%3AS1166)